src/Val.cc

```diff
@@ -3730,9 +3730,18 @@ void VectorVal::ValDescribe(ODesc* d) const
 	d->Add("]");
 	}
 
-ValPtr check_and_promote(ValPtr v, const Type* t, bool is_init,
+ValPtr check_and_promote(ValPtr v, const TypePtr& t, bool is_init,
                          const detail::Location* expr_location)
 	{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+	return check_and_promote(v, t.get(), is_init, expr_location);
+#pragma GCC diagnostic pop
```

Comment:
I suppose this will go away once we removed the deprecated version? Suggest to add add comment explaining.